src/ralph/admin/views/multiadd.py

```diff
@@ -0,0 +1,126 @@
+from django.http import HttpResponseRedirect
+from django.shortcuts import get_object_or_404
+from django.utils.translation import ugettext_lazy as _
+
+from ralph.admin.fields import MultilineField, MultivalueFormMixin
+from ralph.admin.mixins import RalphTemplateView
+from ralph.admin.sites import ralph_site
+
+
+class MultiAddView(RalphTemplateView):
+
+    """Adding multiple assets based on the SN and barcode."""
+
+    template_name = 'admin/multi_add.html'
+
+    def dispatch(
+        self, request, object_pk, model, *args, **kwargs
+    ):
+        self.model = model
+        self.fields = ralph_site._registry[model].multiadd_fields
+        self.obj = get_object_or_404(model, pk=object_pk)
+        return super().dispatch(request, *args, **kwargs)
+
+    def get_form(self):
+        form_kwargs = {}
+        multi_form_fields = {
+            'multivalue_fields': self.fields,
+            'model': self.model
+        }
+        for field in self.fields:
+            multi_form_fields[field] = MultilineField(allow_duplicates=False)
+
+        multi_form = type(
+            'MultiForm', (MultivalueFormMixin, forms.Form), multi_form_fields
+        )
+        if self.request.method == 'POST':
+            form_kwargs['data'] = self.request.POST
+        return multi_form(**form_kwargs)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['form'] = self.get_form()
+        context['obj'] = self.obj
+        context['back_url'] = self.obj.get_absolute_url()
+        return context
+
+    def post(self, request, *args, **kwargs):
+        form = self.get_form()
+        if form.is_valid():
+            try:
+                return self.form_valid(form)
+            except IntegrityError as e:
+                context = self.get_context_data()
+                context['form'] = form
+                form.add_error(None, e.args[1])
+                return self.render_to_response(context)
+        else:
+            return self.form_invalid(form)
+
+    def form_invalid(self, form):
+        context = self.get_context_data()
+        context['form'] = form
+        return self.render_to_response(context)
+
+    @transaction.atomic
+    def form_valid(self, form):
+        args = [form.cleaned_data[field] for field in self.fields]
+        for data in zip(*args):
+            self.obj.baseobject_ptr_id = None
+            self.obj.asset_ptr_id = None
+            self.obj.id = self.obj.pk = None
+            for i, field in enumerate(self.fields):
+                setattr(self.obj, field, data[i])
+
+            try:
+                self.obj.save()
+            except IntegrityError as e:
+                print(forms.forms.NON_FIELD_ERRORS)
+                form._errors[forms.forms.NON_FIELD_ERRORS].append(str(e))
+                return self.form_invalid(form)
+
+        messages.success(
+            self.request, _('Saved many assets successfully with no fuss')
```

Comment:
"Saved {saved_assets} assets" will be better (I know I wrote it, but it was PoC ;))